oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -679,4 +677,112 @@ impl Runtime {
 
         Ok(())
     }
+
+    pub fn new_runtime_proxy(self: Arc<Self>) -> RuntimeProxy {
+        RuntimeProxy {
+            runtime: self.clone(),
+            node_id: self.new_node_reference(),
+        }
+    }
+}
+
+/// A proxy object that binds together a reference to the underlying [`Runtime`] with a single
+/// [`NodeId`].
+///
+/// This can be considered as the interface to the [`Runtime`] that node and pseudo-node
+/// implementations have access to.
+///
+/// Individual methods simply forward to corresponding methods on the underlying [`Runtime`], by
+/// partially applying the first argument.
+#[derive(Clone)]
+pub struct RuntimeProxy {
+    runtime: Arc<Runtime>,
+    node_id: NodeId,
+}
+
+impl RuntimeProxy {
+    /// See [`Runtime::is_terminating`].
+    pub fn is_terminating(&self) -> bool {
+        self.runtime.is_terminating()
+    }
+
+    /// See [`Runtime::remove_node_id`].
+    pub fn exit_node(&self) {
+        self.runtime.remove_node_id(self.node_id)
+    }
+
+    /// See [`Runtime::node_create`].
+    pub fn node_create(
+        &self,
+        module_name: &str,
+        entrypoint: &str,
+        label: &oak_abi::label::Label,
+        channel_read_handle: Handle,
+    ) -> Result<(), OakStatus> {
+        self.runtime.clone().node_create(
+            self.node_id,
+            module_name,
+            entrypoint,
+            label,
+            channel_read_handle,
+        )
+    }
+
+    /// See [`Runtime::new_channel`].
+    pub fn channel_create(&self, label: &oak_abi::label::Label) -> (Handle, Handle) {
+        self.runtime.new_channel(self.node_id, label)
```

Comment:
So all created channels are automatically associated with the node connected to proxy, right?